src/MF_InputShifter/MFInputShifter.cpp

```diff
@@ -10,146 +10,136 @@ inputShifterEvent MFInputShifter::_inputHandler = NULL;
-      trigger(i + (module * 8), currentState & 1);
+            // When triggering event the pin is the actual pin on the chip offset by 8 bits for each
+            // module beyond the first that it's on. The state of the trigger is the bit currently
+            // in position 0 of currentState.
+            trigger(i + (module * 8), currentState & 1);
+        }
+
+        lastState    = lastState >> 1;
+        currentState = currentState >> 1;
     }
-
-    lastState = lastState >> 1;
-    currentState = currentState >> 1;
-  }
 }
 
 // Reads the current state for all connected modules then fires
 // release events for every released button followed by
 // press events for every pressed button.
 void MFInputShifter::retrigger()
 {
-  uint8_t state;
-
-  digitalWrite(_clockPin, HIGH); // Preset clock to retrieve first bit
-  digitalWrite(_latchPin, HIGH); // Disable input latching and enable shifting
-
-  // The current state for all attached modules is stored in the _lastState
-  // array so future update() calls will work off whatever was read by the
-  // retrigger flow.
-  for (int module = 0; module < _moduleCount; module++) {
-    _lastState[module] = shiftIn(_dataPin, _clockPin, MSBFIRST);
-  }
+    uint8_t state;
 
-  digitalWrite(_latchPin, LOW); // disable shifting and enable input latching
+    poll(DONT_TRIGGER);
 
-  // Trigger all the released buttons
-  for (int module = 0; module < _moduleCount; module++) {
-    state = _lastState[module];
+    // Trigger all the released buttons
+    for (int module = 0; module < _moduleCount; module++) {
+        state = _lastState[module];
         for (uint8_t i = 0; i < 8; i++) {
             // Only trigger if the button is in the off position
             if (state & 1) {
-        trigger(i + (module * 8), HIGH);
-      }
-      state = state >> 1;
+                trigger(i + (module * 8), HIGH);
+            }
+            state = state >> 1;
+        }
     }
-  }
 
     // Trigger all the pressed buttons
     for (int module = 0; module < _moduleCount; module++) {
-    state = _lastState[module];
+        state = _lastState[module];
         for (uint8_t i = 0; i < 8; i++) {
             // Only trigger if the button is in the on position
             if (!(state & 1)) {
-        trigger(i + (module * 8), LOW);
-      }
+                trigger(i + (module * 8), LOW);
+            }
 
-      state = state >> 1;
+            state = state >> 1;
+        }
     }
-  }
 }
 
 // Triggers the event handler for the associated input shift register pin,
 // if a handler is registered.
 void MFInputShifter::trigger(uint8_t pin, bool state)
 {
     if (state == LOW && _inputHandler != NULL) {
-    (*_inputHandler)(inputShifterOnPress, pin, _name);
+        (*_inputHandler)(inputShifterOnPress, pin, _name);
     } else if (_inputHandler != NULL) {
-    (*_inputHandler)(inputShifterOnRelease, pin, _name);
-  }
+        (*_inputHandler)(inputShifterOnRelease, pin, _name);
+    }
 }
 
```

Comment:
Would it make sense to use the same code as for the multiplexer?
`if (!_inputHandler) return;
(*_inputHandler)((state ? inputShifterOnRelease : inputShifterOnPress), pin, _name);`

Reply:
Yes it would. Didn't go over this section of code.